### ISSUE TITLE
Validate the join that an `Evaluator`'s constructor performs

### ISSUE THREAD

**Issue by thanasibakis** [CDCgov/cfa-viral-lineage-model#24]
Checking the results of a join can be a lifesaver for otherwise cryptic bugs, even if that's just that the result should be an m x n table, so you `assert res.shape = (expected_rows, expected_columns)`.

_Originally posted by @afmagee42 in https://github.com/CDCgov/cfa-viral-lineage-model/pull/15#discussion_r1711995886_

**Comment by thanasibakis** [comment]
I like this idea, but while implementing it, I learned that LazyFrames don't have shape, and I don't know if it's worth collecting the object in memory just to perform this check?

**Comment by afmagee42** [comment]
@thanasibakis is this going to be solved by #73?

**Comment by thanasibakis** [comment]
Good point, I think it's not solved but made much easier. A quick PR to check the shape inside each Evaluator's constructor should suffice
